### PR TITLE
UNR-1878: Generate Schema Crashes if SchemaDatabase is locked by another process

### PR DESCRIPTION
**Pull request by jherd-improbable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Added SAVE_NoError save flag to SavePackage() to prevent the error being thrown that was locking up the editor when trying to access a locked file.

Modified some of the logic behind error messaging to ensure this case was correctly reported as a failed schema generation.

#### Release note
Generating schema when the schema database is locked by another process will no longer crash the editor.

#### Tests
Successfully reproduced the bug following steps in associated ticket prior to fix. Bug no longer occurs when following the same steps or when attempting to lock the schema database in other ways.

#### Documentation
Bug fix release note.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp [hunk 1]
```diff
@@ -398,7 +398,7 @@ TMap<uint32, FString> CreateComponentIdToClassPathMap()
 	return ComponentIdToClassPath;
 }
 
-void SaveSchemaDatabase()
+bool SaveSchemaDatabase()
```
Comment (Harald-Improbable) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp]: Is this the preferred way to return errors in Unreal? I'm assuming they don't use exceptions but maybe error codes? Output variables? Returning a bool seems very old school to me, but could be how things work in Unreal land.

File: SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp [hunk 2]
```diff
@@ -706,7 +708,10 @@ bool SpatialGDKGenerateSchema()
 	GenerateSchemaFromClasses(TypeInfos, SchemaOutputPath, IdGenerator);
 	GenerateSchemaForSublevels(SchemaOutputPath, IdGenerator);
 	NextAvailableComponentId = IdGenerator.Peek();
-	SaveSchemaDatabase();
+	if (!SaveSchemaDatabase())
+	{
+		return false;
```
Comment (Harald-Improbable) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp]: How is this handled on the calling level? Is there a sensible error message? "Another process is locking this file" kinda stuff?

File: SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp (hunk 1 above)
Comment (jherd-improbable) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp]: I was largely following the convention that the rest of the code there was demonstrating. I agree using bools perhaps isn't the most elegant approach.

File: SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp (hunk 2 above)
Comment (jherd-improbable) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp]: There's an error message output to the log within the function when the actual call to access the file fails. One of my changes actually adjusted the wording of the message to something along the lines you suggest.

Comment (Harald-Improbable) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp]: 👍👍👍👍👍👍👍👍👍

**Pull request merged**